test/index.js

```diff
@@ -284,6 +284,32 @@ describe('koa-body', () => {
       requestSpy = undefined;
     });
 
+    it('should not fail when no request body is provided', (done) => {
+      const echoRouterLayer = router.stack.filter(layer => layer.path === "/echo_body");
+      requestSpy = sinon.spy(echoRouterLayer[0].stack, '0');
+
+      request(http.createServer(app.callback()))
+        .post('/echo_body')
+        .type('application/json')
+        .send(undefined)
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+
+          assert(requestSpy.calledOnce, 'Spy for /echo_body not called');
+          const req = requestSpy.firstCall.args[0].request;
+
+          req.body.should.be.null;
```

Comment:
I don't understand how this is correct or works.
`req.body` is an object with a symbol here, correct?

Reply:
You're right, should have been `.should.not.be`, and on top of that I think it should be a check against `undefined` instead. 

Thanks for pointing this out, as it doesn't do _anything_. I had thought `.null`, or `.String`, etc were getters that caused the assertion to run. All that was happening here was a lookup of the function to trigger assestions. The whole time this should have been `req.body.should.be.Null()`, since the `.null` function wasn't being called, no assertion was running. 

Updated in eb3788a. (Force pushed to fix commit message typo)

Reply:
Thanks @zevisert 
I think the `.should.be.x` pattern is confusing and I had assumed a getter was doing the assertion originally as well.